pkg/common/security/usergroup_test.go

```diff
@@ -64,27 +109,29 @@ func TestGetUserGroup(t *testing.T) {
 	}
 	// click over the clock: if we do not get the cached version the new time will differ from the cache update
 	cachedUG.resolved -= 5
+	testCache.lock.Unlock()
+
 	ug, err = testCache.GetUserGroup("testuser1")
+	testCache.lock.Lock()
 	if err != nil || ug.resolved != cachedUG.resolved {
 		t.Errorf("User 'testuser1' not returned from Cache, resolution time differs: %d got %d (err = %v)", ug.resolved, cachedUG.resolved, err)
 	}
+	testCache.lock.Unlock()
 }
 
 func TestBrokenUserGroup(t *testing.T) {
 	testCache := GetUserGroupCache("test")
 	testCache.resetCache()
 	// test cache should be empty now
-	if len(testCache.ugs) != 0 {
-		t.Fatalf("Cache not empty: %v", testCache.ugs)
-	}
+	assert.Equal(t, 0, testCache.getUGsize(), "Cache is not empty: %v", testCache.getUGmap())
 
 	ug, err := testCache.GetUserGroup("testuser2")
 	if err != nil {
 		t.Error("Lookup should not have failed: testuser2")
 	}
-	if len(testCache.ugs) != 1 {
-		t.Errorf("Cache not updated should have 1 entry %d", len(testCache.ugs))
-	}
+
+	assert.Equal(t, 1, testCache.getUGsize(), "Cache not updated should have 1 entry %d", testCache.getUGmap())
+	testCache.lock.Lock()
```

Comment:
This lock/unlock pair can be removed. `GetUserGroup()` returns a copy and internally uses a mutex to access the map.